sympy/combinatorics/free_groups.py

```diff
@@ -1,18 +1,16 @@
 # -*- coding: utf-8 -*-
 from __future__ import print_function, division
 
-from sympy.core.basic import Basic
+from sympy.core.backend import Basic, Symbol, S
 from sympy.core.compatibility import is_sequence, as_int, string_types
 from sympy.core.expr import Expr
```

Comment:
We should make a simple `Expr` class like
```
cdef class Expr(Basic)
    pass
```
and then change other classes like `class Add(Basic)` to `class Add(Expr)`